src/com/lms/model/Branch.java

```diff
@@ -2,42 +2,92 @@
 
 import java.util.Objects;
 
+/**
+ * A branch of a library.
+ *
+ * @author Salem Ozaki
+ * @author Jonathan Lovelace
+ */
 public class Branch {
+	/**
+	 * The ID number used to identify this branch in the database.
+	 */
 	private final int id;
+	/**
+	 * The name of the branch.
+	 */
 	private String name;
+	/**
+	 * The address of the branch.
+	 */
 	private String address;
-	
-	public Branch(int id, String name, String address) {
+
+	/**
+	 * To construct a branch object, callers must supply its ID number, name, and
+	 * address.
+	 *
+	 * @param id      the ID number to identify this branch.
+	 * @param name    The name of the branch
+	 * @param address The address of the branch
+	 */
+	public Branch(final int id, final String name, final String address) {
 		this.id = id;
 		this.name = name;
 		this.address = address;
 	}
 
+	/**
+	 * Get the name of the branch, which will not be null.
```

Comment:
Actually, it can be null. In fact Null is a default value. Same with address

Reply:
I admit that in these comments I wrote more prescriptively than descriptively. In my implementation I will add code to any constructors that currently don't set fields to make sure that no field ends up as null by default, and I will make sure that SQL NULL gets translated to the empty string (and vice versa) at the lowest possible level of the DAO.

The general principle I try to code by is that if an interface I am creating is intended to support `null`, it should support it *explicitly*, via `java.util.Optional` or one of the `@Nullable` annotations (there are two competing versions from the makers of the Eclipse IDE, at least one from JetBrains, one from the makers of the FindBugs static analysis tool, and at least one other one ...), and if not, the fact that it's not intended to support `null` should be documented.

Reply:
Okay, that makes sense.